wurlitzer.py

```diff
@@ -194,18 +198,21 @@ def forwarder():
                         flush_queue.put('flush')
                         flush_interval = self.flush_interval
                         continue
-                for pipe in r:
-                    if pipe == self._control_r:
+
+                for fd, flags in events:
+                    if fd == self._control_r:
                         draining = True
                         os.close(self._control_r)
                         pipes.remove(self._control_r)
+                        poller.unregister(self._control_r)
                         continue
-                    name = names[pipe]
-                    data = os.read(pipe, 1024)
+                    name = names[fd]
+                    data = os.read(fd, 1024)
                     if not data:
                         # pipe closed, stop polling it
-                        os.close(pipe)
-                        pipes.remove(pipe)
+                        os.close(fd)
+                        pipes.remove(fd)
+                        poller.unregister(fd)
```

Comment:
Maybe call unregister before close? Just in case unregister checks that the fd is valid.

Reply:
got it good catch, will change that